### PR TITLE
SPA Animal Mimicry

### PR DESCRIPTION
**Pull request by pheonixstorm**
Up for review. The code wasn't hard to deal with and the protomechs were simple to add along after looking over the unit files.

@arlith if you are able to review today ping me in slack if you want me to push an updated history.txt

### CODE REVIEW

**Comment by HammerGS** [comment]
This one is going to conflict with IO branch when merged in.  Especially the canon quirks file since it's completely different.  I'd leave that file out, and then I'll manually add the quirk to the totem mechs after

**Comment by pheonixstorm** [comment]
Ok, will revert the changes.

File: megamek/src/megamek/common/Terrain.java [hunk 1]
```diff
@@ -399,6 +399,14 @@ public int movementCost(Entity e) {
                                 && (moveMode == EntityMovementMode.INF_LEG))) {
                     mp -= 1;
                 }
+                if (e.getCrew().getOptions().booleanOption(OptionsConstants.PILOT_ANIMAL_MIMIC)) {
+                    if (e.entityIsQuad()) {
+                        mp -= 1;
+                    }
+                    if ((moveMode == EntityMovementMode.BIPED) && e.hasQuirk("animalistic")) {
+                        mp -= 1;
+                    }
```
Comment (dericpage) [inline megamek/src/megamek/common/Terrain.java]: You should be able to combine these if statements.

File: megamek/src/megamek/common/Terrain.java [hunk 2]
```diff
@@ -412,6 +420,14 @@ public int movementCost(Entity e) {
...
```
Comment (dericpage) [inline megamek/src/megamek/common/Terrain.java]: Same here.

File: megamek/src/megamek/common/Terrain.java (hunk 1 above)
Comment (pheonixstorm) [reply megamek/src/megamek/common/Terrain.java]: so
if quad or biped with animalistic mp -= 1?
I wasn't sure about doing it that way as I initially wanted to keep the quads and bipeds separate.

Comment (dericpage) [reply megamek/src/megamek/common/Terrain.java]: `if (e.entityIsQuad() || ((EntityMovementMode.BIPED == moveMode) && e.hasQuirk(OptionsConstants.QUIRK_POS_ANIMALISTIC)) {`

**Comment by pheonixstorm** [comment]
Should be ready now

**Comment by HammerGS** [comment]
Merged

**Pull request merged**